Scripts/utils/read_csv_file.py

```diff
@@ -68,17 +68,17 @@ def read_csv_file(data_dir: str,
     if data.index.has_duplicates:
         raise IndexError("Index in file {} has duplicates".format(path))
     if zone_numbers is not None:
-        if not data.index.is_monotonic:
+        if not data.index.is_monotonic_increasing:
             data.sort_index(inplace=True)
             log.warn("File {} is not sorted in ascending order".format(path))
         map_path = os.path.join(data_dir, "zone_mapping.txt")
         if os.path.exists(map_path):
             log_path = map_path
-            mapping = pandas.read_csv(map_path, delim_whitespace=True).squeeze()
+            mapping = pandas.read_csv(map_path, sep='\s+').squeeze()
             if "total" in data.columns:
                 # If file contains total and shares of total,
                 # shares are aggregated as averages with total as weight
-                data = data.groupby(mapping).agg(avg, weights=data["total"])
+                data = data.groupby(mapping).agg(lambda ser: avg(ser, weights=data["total"]))
```

Comment:
What was the problem here? I see no indications in the pandas documentation that feeding regular functions into `agg` would have been deprecated.

Reply:
I could not make the avg function work without this. It seems that avg function expects to get a Series as an argument, but will get a DataFrame object instead. For some reason adding lambda here fixes that. There might be a more elegant way to fix this.